h2o-algos/src/main/java/hex/tree/sdt/SDT.java

```diff
@@ -0,0 +1,368 @@
+            limitsQueue.add(null);
+            makeListFromNode(zeroRatio, nodeIndex);
+            return;
+        }
+
+        Histogram histogram = new Histogram(_train, actualLimits, BinningStrategy.EQUAL_WIDTH/*, minNumSamplesInBin - todo consider*/);
+
+        SplitInfo bestSplitInfo = findBestSplit(histogram);
+        // if no split could be found, make a list from current node
+        if (bestSplitInfo == null) {
+            // add imaginary left and right children to imitate right tree structure
+            // left child
+            limitsQueue.add(null);
+            // right child
+            limitsQueue.add(null);
+            makeListFromNode(zeroRatio, nodeIndex);
+            return;
+        }
+
+        // flag that node is not a list
+        _tree[nodeIndex][0] = 0;
+        _tree[nodeIndex][1] = bestSplitInfo._splitFeatureIndex;
+        _tree[nodeIndex][2] = bestSplitInfo._threshold;
+
+        DataFeaturesLimits limitsLeft = actualLimits.updateMax(bestSplitInfo._splitFeatureIndex, bestSplitInfo._threshold);
+        DataFeaturesLimits limitsRight = actualLimits.updateMin(bestSplitInfo._splitFeatureIndex, bestSplitInfo._threshold);
+        Log.debug("root: " + countClasses(actualLimits) + ", left: " + countClasses(limitsLeft) +
+                ", right: " + countClasses(limitsRight) + ", best feature: " + bestSplitInfo._splitFeatureIndex
+                + ", threshold: " + bestSplitInfo._threshold);
+
+        Log.debug("feature: " + bestSplitInfo._splitFeatureIndex + ", threshold: " + bestSplitInfo._threshold);
+        Log.debug("Left min-max: " + limitsLeft.getFeatureLimits(bestSplitInfo._splitFeatureIndex)._min +
+                " " + limitsLeft.getFeatureLimits(bestSplitInfo._splitFeatureIndex)._max);
+        Log.debug("Right min-max: " + limitsRight.getFeatureLimits(bestSplitInfo._splitFeatureIndex)._min +
+                " " + limitsRight.getFeatureLimits(bestSplitInfo._splitFeatureIndex)._max);
+
+        // store limits for left child
+        limitsQueue.add(limitsLeft);
+        // store limits for right child
+        limitsQueue.add(limitsRight);
+    }
+
+
+    /**
+     * Compute initial features limits.
+     *
+     * @return features limits
+     */
+    public static DataFeaturesLimits getInitialFeaturesLimits(Frame data) {
+        return new DataFeaturesLimits(
+                IntStream.range(0, data.numCols() - 1 /*exclude the last prediction column*/)
+                        .mapToObj(i -> data.vec(i))
+                        // decrease min as the minimum border is always excluded and real min value could be lost
+                        .map(v -> new FeatureLimits(v.min() - EPSILON, v.max()))
+                        .collect(Collectors.toList()));
+    }
+
+
+    private class SDTDriver extends Driver {
+
+        @Override
+        public void computeImpl() {
+            _model = null;
+            try {
+                init(false);
+                if (error_count() > 0) {
```

Comment:
Please change init(false) to init(true) and add the following checks:

1. If training frame include categorical columns, throw an error;
2. if response is not binary, throw an error;